src/hooks/use-deposit.js

```diff
@@ -72,7 +72,11 @@ function useDeposit(wallet, network, tokenInfo) {
     }
 
     const wc = newDeposits[0].withdrawal_credentials
-    const isBatch = newDeposits.every(d => d.withdrawal_credentials === wc)
+    let isBatch = false;
+
+    if (newDeposits.length > 1) {
+      isBatch = newDeposits.every(d => d.withdrawal_credentials === wc)
+    }
```

Comment:
The point is that this line
```js
newDeposits.every(d => d.withdrawal_credentials === wc)
```
always returns `true` even for single deposit in a list. So, this check is necessary to not process single deposits as batches.

Reply:
I feel like there's a bug with the batching thing, because the code implies that it's possible to deposit stake for different withdrawal credentials, but I don't see how this could happen here: https://github.com/gnosischain/deposit-contract/blob/master/contracts/SBCDepositContract.sol#L100-L124

I'll have to investigate this a bit more.

Reply:
Actually @4rgon4ut you should revert this. The naming is a bit confusing, but with one deposit it should go in the batch part. The `isBatch === false` part is when there are deposits with different withdrawal credentials, as these actually require multiple transactions.

So please revert this change 🙂

Reply:
@filoozom done